climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define WildFire class.
+"""
+
+__all__ = ['WildFire']
+
+import logging
+import warnings
+import itertools
+from datetime import date
+from functools import partial
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy import sparse
+from sklearn.neighbors import BallTree
+from sklearn.cluster import DBSCAN
+import numba
+import geopandas as gpd
+from shapely.geometry import Point
+from shapely.ops import transform
+import pyproj
+
+from climada.hazard.centroids.centr import Centroids
+from climada.hazard.base import Hazard
+from climada.hazard.tag import Tag as TagHazard
+from climada.util.constants import ONE_LAT_KM, DEF_CRS
+from climada.util.dates_times import str_to_date
+
+from climada.util.alpha_shape import alpha_shape, plot_polygon
+warnings.simplefilter(action='ignore', category=FutureWarning)
+
+LOGGER = logging.getLogger(__name__)
+
+HAZ_TYPE = 'WF'
+""" Hazard type acronym for Wild Fire, might be changed to WFseason or WFsingle """
+
+CLEAN_THRESH = 30
+""" Minimal confidence value for the data from MODIS instrument to be use as input"""
+
+RES_DATA = 1.0
+""" Resolution of the data if no data origin provided (km) """
+
+BLURR_STEPS = 4
+""" steps with exponential decay for fire propagation matrix """
+
+class WildFire(Hazard):
+
+    """Contains wild fires.
+
+    Wildfires comprise the challange that the definition of an event is unclear.
+    Reporting standards vary accross regions and over time. Hence, to have
+    consistency, we consider an event as a whole fire season. This allows
+    consistent risk assessment across the global and over time. Events that
+    refer to a fire season have the tag 'WFseason'.
+
+    In order to perform concrete case studies or calibrate impact functions,
+    events can be displayed as single fires. In that case they have the tag
+    'WFsingle'.
+
+
+    Attributes:
+        date_end ((np.array): integer date corresponding to the proleptic
+            Gregorian ordinal, where January 1 of year 1 has ordinal 1
+            (ordinal format of datetime library))
+
+        n_fires (np.array): number of single fires in a fire season
+
+    """
+
+    days_thres_firms = 2
+    """ Minimum number of days to consider different fires """
+
+    clus_thres_firms = 15
+    """ Clustering factor which multiplies instrument resolution """
+
+    remove_minor_fires_firms = True
+    """ removes FIRMS fires below defined theshold of entries"""
+
+    minor_fire_thres_firms = 3
+    """ number of FIRMS entries required to be considered a fire """
+
+    prop_proba = 0.21
+    """ probability of fire propagation """
+
+    max_it_propa = 500000
+    """ maximum propgation iterations for a probabilistic fire """
```

Comment:
Are these constant that should not be changed by a user, or are these constants that might change depending on the use case? If it is the latter, please remove them here and define them in the configuration file.

Why are these values defined here instead of global constants?

Reply:
Ok! Then, these should either be model inputs (i.e., function parameters), or be included in the configuration file.

Reply:
how do I best go about it then? I don't really want to do them into a config file, but I also don't want to give more function parameters to the set_[--]_wildfire functions?

Reply:
Is there a concrete reason why not put them into the config file? Either these are input variables for the given method, or they are configurable constants (config-file), or they are non-configurable constants (constants in this module). As far as I understand it here, these are configurable constants right? Or are they rather input variables?  To avoid having to many input variables, you could also make these attributes of your class if they are really characterising the hazard.

Reply:
would it make sense to write two functions (_set_params_firms() and _set_params_proba()) which in the default assign these values as attributes of the class but makes it easier to modify them?

Reply:
Interesting idea. From my experience however, I would not do that. But I am not sure. @emanuel-schmid could you please answer to that?

Reply:
From the interface point of view (without knowing what is going on): yes, makes sense. Somewhat.
But - why hide it? My suggestion:
create two [data classes](https://docs.python.org/3.8/library/dataclasses.html) `FirmsParams` and `PobaParams` instead.
Data types, default values, all there and everyting right there in your face. IDE knows all about it, attributes have fixed names and you don't 'pollute' the name space with global variables, allmost no coding, nicely bundled, easy to understand. etc. etc.
Very cool stuff. :grin:

Reply:
Thanks emanuel, I really like this option.
I've adjusted the code eaaf3b61